pkg/scaffold/v2/main.go

```diff
@@ -72,14 +72,28 @@ func init() {
 	// %s
 }
 
+// Persmissions to do leader election.
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=configmaps/status,verbs=get;update;patch
+
 func main() {
 	var metricsAddr string
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.Parse()
 
 	ctrl.SetLogger(zap.Logger(true))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{ Scheme: scheme, MetricsBindAddress: metricsAddr })
+	leaderElectionNamespace := "default"
+	if len(os.Getenv("POD_NAMESPACE")) != 0 {
```

Comment:
IIRC we do this automatically when leader election is enabled -- no need to re-do it here.

Reply:
LE in CR current default ns to `default` and configmap name to `controller-leader-election-helper`.
Ref: https://github.com/kubernetes-sigs/controller-runtime/issues/445

Reply:
After thinking it more, we realize we should set different [`LeaderElectionID`](https://github.com/kubernetes-sigs/controller-runtime/blob/687d7fc8ef4fa76256188c9046afe8bbf30acbf8/pkg/manager/manager.go#L116:2) here. Probably something like `<kind>-controller-leader-election-helper`.
So users can deploy 2 controllers in the same namespace without leader election object conflicts.

Reply:
it uses the current namespace: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/leaderelection/leader_election.go#L63